### PR TITLE
deprecate document_status field

### LINKED ISSUE #1123: Deprecate DocumentStatus field in favour of DeletedAt
The `DocumentStatus` in our models is becoming an issue & it hard to keep track of alongside `DeletedAt`, it should be deprecated so we only use `DeletedAt` to identify deleted records.

### PR DESCRIPTION
**Pull request by danvixent**
fixes #1123

### CODE REVIEW

File: datastore/mongo/api_key.go [hunk 1]
```diff
@@ -65,8 +65,8 @@ func (db *apiKeyRepo) FindAPIKeyByMaskID(ctx context.Context, maskID string) (*d
 	apiKey := new(datastore.APIKey)
 
 	filter := bson.M{
-		"mask_id":         maskID,
-		"document_status": datastore.ActiveDocumentStatus,
+		"mask_id":    maskID,
+		"deleted_at": nil,
```
Comment (Dotunj) [inline datastore/mongo/api_key.go:69]: The store already handles this, do we still need to pass it here?

File: datastore/db.go [hunk 2]
```diff
@@ -421,10 +415,7 @@ func (d *MongoStore) DeleteOne(ctx context.Context, filter bson.M, hardDelete bo
 		return err
 
 	} else {
-		payload := bson.M{
-			"deleted_at":      primitive.NewDateTimeFromTime(time.Now()),
-			"document_status": DeletedDocumentStatus,
-		}
+		payload := bson.M{"deleted_at": primitive.NewDateTimeFromTime(time.Now())}
```
Comment (Subomi) [inline datastore/db.go:418]: @danvixent We didn't apply this to `DeleteMany`

**Review by Subomi — CHANGES_REQUESTED** [review]
@danvixent FYI

File: datastore/mongo/api_key.go (hunk 1 above)
Comment (Subomi) [reply datastore/mongo/api_key.go:69]: I think this affects almost all repository methods.

File: datastore/db.go (hunk 2 above)
Comment (danvixent) [reply datastore/db.go]: fixed

File: datastore/mongo/application.go [hunk 3]
```diff
@@ -318,7 +314,7 @@ func (db *appRepo) assertUniqueAppTitle(ctx context.Context, app *datastore.Appl
 	f := bson.M{
 		"uid":      bson.M{"$ne": app.UID},
 		"title":    app.Title,
-		"group_id": groupID,
+		"group_id": groupID, // TODO: filter out deleted_at
```
Comment (jirevwe) [inline datastore/mongo/application.go:317]: @danvixent is this TODO still needed?

File: datastore/mongo/application_test.go [hunk 4]
```diff
@@ -28,29 +29,29 @@ func Test_UpdateApplication(t *testing.T) {
...
 	newTitle := "Newer name"
-
 	app.Title = newTitle
 
 	require.NoError(t, appRepo.UpdateApplication(context.Background(), app, app.GroupID))
 
 	newApp, err := appRepo.FindApplicationByID(context.Background(), app.UID)
 	require.NoError(t, err)
-
+	fmt.Printf("tt %+v", newApp)
```
Comment (jirevwe) [inline datastore/mongo/application_test.go:48]: Don't think this is needed

Comment (danvixent) [reply datastore/mongo/application_test.go]: fixed

File: datastore/mongo/application.go (hunk 3 above)
Comment (danvixent) [reply datastore/mongo/application.go]: fixed

File: datastore/mongo/mongo_test.go [hunk 5]
```diff
@@ -15,11 +14,10 @@ import (
 )
 
 func getDSN() string {
-	return os.Getenv("TEST_MONGO_DSN")
+	return "mongodb://localhost:27017/testdb"
```
Comment (jirevwe) [inline datastore/mongo/mongo_test.go:17]: we need to undo this before merging

Comment (danvixent) [reply datastore/mongo/mongo_test.go]: fixed

**Review by jirevwe — APPROVED** [review]

**Pull request merged**